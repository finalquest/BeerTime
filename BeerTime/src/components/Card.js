import React from 'react';
import {
  View,
  Image,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import PropTypes from 'prop-types';

import { BeerProptype } from '../utils/PropTypes';
import Label from './Label';
import PropertyLabel from './PropertyLabel';

import Empty from '../../res/starEmpty.png';
import Star from '../../res/star.png';


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 120,
    backgroundColor: 'white',
    marginHorizontal: 15,
    marginVertical: 10,
    padding: 10,
    borderRadius: 4,
    borderColor: 'lightgray',
    borderWidth: 1,
  },
  imageContainer: {
    flex: 0.2,
    marginLeft: 5,
  },
  image: {
    resizeMode: 'contain',
    flex: 1,
  },
  bodyContainer: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    marginHorizontal: 15,
  },
  name: {
    fontSize: 20, textAlign: 'left',
  },
  icon: {
    alignSelf: 'center',
    margin: 10,
    width: 25,
    height: 25,
    resizeMode: 'contain',
  },
});


const Card = ({ beer, onItemSelected, addToFavs }) => {
  const {
    image_url: imageUrl, name,
    tagline, first_brewed: firstBrewed, id, selected,
  } = beer;
  const imageSource = selected ? Star : Empty;
  return (
    <TouchableOpacity style={styles.container} onPress={() => onItemSelected(beer)}>
      <View style={styles.imageContainer}>
        <Image
          style={styles.image}
          source={{ uri: imageUrl }}
        />
      </View>
      <View style={styles.bodyContainer}>
        <Label style={styles.name} ellipsizeMode="tail" numberOfLines={2}>{name}</Label>
        <Label style={{ fontSize: 11 }}>{tagline}</Label>
        <PropertyLabel
          name="First brew date"
          style={{ marginVertical: 10 }}
          value={firstBrewed}
          separator=":"
        />
      </View>
      <TouchableOpacity onPress={() => { addToFavs(id); }}>
        <Image source={imageSource} style={styles.icon} />
      </TouchableOpacity>
    </TouchableOpacity>
  );
};
Card.propTypes = {
  beer: BeerProptype.isRequired,
  onItemSelected: PropTypes.func.isRequired,
  addToFavs: PropTypes.func.isRequired,
};
Card.defaultProps = {
};

export default Card;
