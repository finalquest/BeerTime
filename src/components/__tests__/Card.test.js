import React from 'react';
import { shallow } from 'enzyme';

import Card from '../Card';

describe('Card test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Card beer={{ image_url: 'testUrl' }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call itemSelected', () => {
    const itemSelected = jest.fn();
    const wrapper = shallow(<Card
      beer={{ image_url: 'testUrl', id: 1 }}
      onItemSelected={itemSelected}
    />);
    wrapper.find('TouchableOpacity').at(0).prop('onPress')();
    expect(itemSelected).toBeCalledWith({ image_url: 'testUrl', id: 1 });
  });

  it('should call addToFavs', () => {
    const addToFavs = jest.fn();
    const wrapper = shallow(<Card
      beer={{ image_url: 'testUrl', id: 1 }}
      addToFavs={addToFavs}
    />);
    wrapper.find('TouchableOpacity').at(1).prop('onPress')();
    expect(addToFavs).toBeCalledWith(1);
  });
});
