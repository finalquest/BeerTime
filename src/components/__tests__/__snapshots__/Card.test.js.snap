// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suite renders as expected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "lightgray",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 120,
      "marginHorizontal": 15,
      "marginVertical": 10,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.2,
        "marginLeft": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "testUrl",
        }
      }
      style={
        Object {
          "flex": 1,
          "resizeMode": "contain",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "justifyContent": "flex-start",
        "marginHorizontal": 15,
      }
    }
  >
    <Label
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "fontSize": 20,
          "textAlign": "left",
        }
      }
      uppercase={false}
    />
    <Label
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 11,
        }
      }
      uppercase={false}
    />
    <PropertyLabel
      Component={[Function]}
      ellipsizeMode="tail"
      fontSize={14}
      name="First brew date"
      nameStyles={
        Object {
          "fontWeight": "400",
          "marginRight": 3,
          "textAlign": "center",
        }
      }
      numberOfLines={1}
      separator=":"
      style={
        Object {
          "marginVertical": 10,
        }
      }
      valueStyles={
        Object {
          "flexShrink": 1,
          "textAlign": "center",
        }
      }
    />
  </View>
  <TouchableOpacity
    onPress={[Function]}
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 25,
          "margin": 10,
          "resizeMode": "contain",
          "width": 25,
        }
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;
